up42/storage.py

```diff
@@ -78,6 +78,34 @@ def _query_paginated(
             results_list += response_json["content"]
         return results_list[:limit]
 
+    def _query_paginated_stac_search(
+        self, 
+        url: str, 
+        stac_search_parameters: dict,
+        ) -> list:
+        """
+        Helper to fetch list of items in paginated stac search endpoind, e.g. stac search assets.
+
+        Args:
+            url (str): The base url for paginated endpoint.
+            stac_search_parameters (dict): the parameters required for stac search
+
+        Returns:
+            List of storage STAC results features.
+        """
+        response_features = []
+        while True:
```

Comment:
`while` loop obviously dangerous. Even though the logic seems save, we could in theory add another precaution:

Since we have `max_items` and `limit` in the search_parameters, we know that there can be a maximum of `limit/max_items` requests.